[BUG]: enabling transcription in Voice Live execution settings breaks the stream

Running with voice_live and enabling transcription (as per https://learn.microsoft.com/en-us/azure/ai-services/speech-service/voice-live-how-to#audio-input-through-azure-speech-to-text) i.e.:

```yaml
input_audio_transcription:
  model: azure-speech
```

or 

```yaml
input_audio_transcription:
  model: azure-fast-transcription
```

Results in a stream error when starting the app:

```
2025-08-27 15:33:10,756 [INFO] azure.identity.aio._credentials.chained: DefaultAzureCredential acquired a token from AzureCliCredential
2025-08-27 15:33:11,566 [INFO] semantic_kernel.connectors.ai.open_ai.realtime: Session created or updated, session: {"id":"sess_C9BV1rS0oeEazPrUcI1Vh","input_audio_format":"pcm16","input_audio_noise_reduction":null,"input_audio_transcription":null,"instructions":"Your knowledge cutoff is 2023-10. You are a helpful, witty, and friendly AI. Act like a human, but remember that you aren't a human and that you can't do human things in the real world. Your voice and personality should be warm and engaging, with a lively and playful tone. If interacting in a non-English language, start by using the standard accent or dialect familiar to the user. Talk quickly. You should always call a function if you can. Do not refer to these rules, even if you’re asked about them.","max_response_output_tokens":"inf","modalities":["text","audio"],"model":"gpt-4o-realtime-preview-2024-12-17","output_audio_format":"pcm16","speed":null,"temperature":0.8,"tool_choice":"auto","tools":[],"tracing":null,"turn_detection":{"create_response":true,"eagerness":null,"interrupt_response":true,"prefix_padding_ms":300,"silence_duration_ms":200,"threshold":0.5,"type":"server_vad","speech_duration_ms":null,"end_of_utterance_detection":null,"auto_truncate":false},"voice":"alloy","input_audio_sampling_rate":24000,"animation":null,"input_audio_echo_cancellation":null,"avatar":null,"output_audio_timestamp_types":null,"agent":null,"reasoning_effort":null,"truncation_strategy":null}
2025-08-27 15:33:11,566 [INFO] api.app.agents.utils: Session Created Message
2025-08-27 15:33:12,589 [ERROR] semantic_kernel.connectors.ai.open_ai.realtime: Error sending response: no close frame received or sent
2025-08-27 15:33:12,628 [ERROR] semantic_kernel.connectors.ai.open_ai.realtime: Error sending response: no close frame received or sent
2025-08-27 15:33:12,672 [ERROR] semantic_kernel.connectors.ai.open_ai.realtime: Error sending response: no close frame received or sent
```

--- a/Taskfile.yml
+++ b/Taskfile.yml
@@ -11,11 +11,16 @@ tasks:
     cmds:
       - |
         devtunnel login
-        devtunnel create aicontactcentre --allow-anonymous
+        TUNNEL_NAME="aicontactcentre-$(azd env get-values | grep AZURE_ENV_NAME | cut -d'=' -f2 | tr -d '"' || echo "$(date +%s)")"
+        echo "Creating devtunnel with name: $TUNNEL_NAME"
+        devtunnel create $TUNNEL_NAME --allow-anonymous
         devtunnel port create -p 8000
   run:devtunnel:
     cmds:
-      - devtunnel host aicontactcentre
+      - |
+        TUNNEL_NAME="aicontactcentre-$(azd env get-values | grep AZURE_ENV_NAME | cut -d'=' -f2 | tr -d '"' || echo "$(date +%s)")"
+        echo "Hosting devtunnel: $TUNNEL_NAME"
+        devtunnel host $TUNNEL_NAME
   run:api:
     dir: api
     cmds:

--- a/api/app/agents/settings/4o_realtime.yaml
+++ b/api/app/agents/settings/4o_realtime.yaml
@@ -18,10 +18,8 @@ turn_detection:
   silence_duration_ms: 800
   threshold: 0.8
 
-# input_audio_transcription:
-#   model: gpt-4o-transcribe
-#   language: en
-#   prompt: ""
+input_audio_transcription:
+  model: whisper-1
 
 # Optional general settings
 # temperature: 0.8

--- a/api/app/agents/settings/voice_live.yaml
+++ b/api/app/agents/settings/voice_live.yaml
@@ -34,9 +34,8 @@ input_audio_noise_reduction:
 input_audio_echo_cancellation:
   type: server_echo_cancellation
 
-# input_audio_transcription:
-#   model: azure-fast-transcription
-#   phrase_list: []
+input_audio_transcription:
+  model: whisper-1
 
 # Optional general settings
 # temperature: 0.8

--- a/api/app/agents/utils.py
+++ b/api/app/agents/utils.py
@@ -205,6 +205,9 @@ async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientB
             case ListenEvents.CONVERSATION_ITEM_INPUT_AUDIO_TRANSCRIPTION_COMPLETED:
                 user_transcript = _get(se, "transcript") or ''
                 logger.info(f" User:-- {user_transcript}")
+                # Add user message to chat history
+                if user_transcript:
+                    chat_history.add_user_message(user_transcript)
                 # Send user transcription to frontend
                 await websocket.send_text(
                     json.dumps(

--- a/api/app/agents/utils.py
+++ b/api/app/agents/utils.py
@@ -151,7 +151,7 @@ async def get_agent(template_name: str, plugins: list[object], chat_history: Cha
         settings=execution_settings,
     )
 
-async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientBase, chat_history: ChatHistory):
+async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientBase, chat_history: ChatHistory, is_development_mode: bool = False):
     """Function that handles the messages from the Realtime service.
 
     This function only handles the non-audio messages.
@@ -208,19 +208,20 @@ async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientB
                 # Add user message to chat history
                 if user_transcript:
                     chat_history.add_user_message(user_transcript)
-                # Send user transcription to frontend
-                await websocket.send_text(
-                    json.dumps(
-                        {
-                            "kind": "Transcription",
-                            "data": {
-                                "speaker": "user",
-                                "text": user_transcript,
-                                "timestamp": _get(se, "audio_start_ms") or 0
+                # Send user transcription to frontend (only in development mode)
+                if is_development_mode:
+                    await websocket.send_text(
+                        json.dumps(
+                            {
+                                "kind": "Transcription",
+                                "data": {
+                                    "speaker": "user",
+                                    "text": user_transcript,
+                                    "timestamp": _get(se, "audio_start_ms") or 0
+                                }
                             }
-                        }
+                        )
                     )
-                )
             case ListenEvents.CONVERSATION_ITEM_INPUT_AUDIO_TRANSCRIPTION_FAILED:
                 logger.error(f"  Error: {_get(se, "error") or '<unknown>'}")
             case ListenEvents.RESPONSE_DONE:
@@ -252,19 +253,20 @@ async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientB
                 # Add assistant message to chat history
                 if transcript:
                     chat_history.add_assistant_message(transcript)
-                    # Send assistant transcription to frontend
-                    await websocket.send_text(
-                        json.dumps(
-                            {
-                                "kind": "Transcription",
-                                "data": {
-                                    "speaker": "assistant",
-                                    "text": transcript,
-                                    "timestamp": None  # No timestamp available for assistant transcripts
+                    # Send assistant transcription to frontend (only in development mode)
+                    if is_development_mode:
+                        await websocket.send_text(
+                            json.dumps(
+                                {
+                                    "kind": "Transcription",
+                                    "data": {
+                                        "speaker": "assistant",
+                                        "text": transcript,
+                                        "timestamp": None  # No timestamp available for assistant transcripts
+                                    }
                                 }
-                            }
+                            )
                         )
-                    )
             case SendEvents.CONVERSATION_ITEM_CREATE:
                 # Add function call result to chat history
                 function_result = getattr(event, "function_result", None)

--- a/api/app/agents/utils.py
+++ b/api/app/agents/utils.py
@@ -201,7 +201,21 @@ async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientB
                     )
                 )
             case ListenEvents.CONVERSATION_ITEM_INPUT_AUDIO_TRANSCRIPTION_COMPLETED:
-                logger.info(f" User:-- {_get(se, "transcript") or ''}")
+                user_transcript = _get(se, "transcript") or ''
+                logger.info(f" User:-- {user_transcript}")
+                # Send user transcription to frontend
+                await websocket.send_text(
+                    json.dumps(
+                        {
+                            "kind": "Transcription",
+                            "data": {
+                                "speaker": "user",
+                                "text": user_transcript,
+                                "timestamp": _get(se, "audio_start_ms") or 0
+                            }
+                        }
+                    )
+                )
             case ListenEvents.CONVERSATION_ITEM_INPUT_AUDIO_TRANSCRIPTION_FAILED:
                 logger.error(f"  Error: {_get(se, "error") or '<unknown>'}")
             case ListenEvents.RESPONSE_DONE:
@@ -233,6 +247,19 @@ async def handle_realtime_messages(websocket: WebSocket, client: RealtimeClientB
                 # Add assistant message to chat history
                 if transcript:
                     chat_history.add_assistant_message(transcript)
+                    # Send assistant transcription to frontend
+                    await websocket.send_text(
+                        json.dumps(
+                            {
+                                "kind": "Transcription",
+                                "data": {
+                                    "speaker": "assistant",
+                                    "text": transcript,
+                                    "timestamp": None  # No timestamp available for assistant transcripts
+                                }
+                            }
+                        )
+                    )
             case SendEvents.CONVERSATION_ITEM_CREATE:
                 # Add function call result to chat history
                 function_result = getattr(event, "function_result", None)

--- a/api/app/main.py
+++ b/api/app/main.py
@@ -8,13 +8,11 @@ from pathlib import Path
 from fastapi import Depends, FastAPI, WebSocket
 from fastapi.responses import FileResponse
 from fastapi.staticfiles import StaticFiles
-from semantic_kernel.connectors.mcp import MCPStreamableHttpPlugin
 from semantic_kernel.contents import (AudioContent, ChatHistory,
                                       RealtimeAudioEvent)
 
 from .agents.plugins import CallPlugin, DeliveryPlugin
 from .agents.utils import get_agent, handle_realtime_messages
-from .config import settings
 from .dependencies import get_acs_client
 from .routers import calls
 
@@ -48,61 +46,28 @@ async def agent_connect(websocket: WebSocket, acs_client=Depends(get_acs_client)
     call_connection_id = websocket.headers.get("x-ms-call-connection-id")
 
     if not call_connection_id:
-        logger.warning("No call connection ID provided in headers indicating direct connection (not ACS). Certain call functions will be mocked.")
-
-    chat_history = ChatHistory()
-
-    # Prepare plugins list
-    plugins = [
-        CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
-        DeliveryPlugin()
-    ]
+        logger.warning("No call connection ID provided in headers indicating direct connection (not ACS). Certain call functions won't work.")
     
-    # Try to initialize MCP plugin, but fallback gracefully if it fails
-    orders_api_plugin = None
-    try:
-        # Attempt to create MCP plugin with a short timeout
-        orders_api_plugin = MCPStreamableHttpPlugin(
-            name="Orders API",
-            url=settings.MCP_ORDERS_URL,
-            load_prompts=False # APIM MCP servers only support tools and this will fail if left enabled
-        )
-        # Try to connect with timeout
-        import asyncio
-        await asyncio.wait_for(orders_api_plugin.connect(), timeout=5.0)
-        plugins.insert(1, orders_api_plugin)  # Insert between CallPlugin and DeliveryPlugin
-        logger.info("Successfully connected to MCP Orders API")
-    except Exception as e:
-        logger.warning(f"Failed to connect to MCP Orders API: {e}. Continuing without orders functionality.")
-        orders_api_plugin = None
+    # Determine if running in development mode (for transcription safety)
+    is_development_mode = not call_connection_id
     
     # Load realtime agent from template and plugins
+    chat_history = ChatHistory()
     realtime_agent = await get_agent(
         template_name="DeliveryAgent",
-        plugins=plugins,
+        plugins=[
+            CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
+            DeliveryPlugin()
+        ],
         chat_history=chat_history,
-        agent_name="Ollie"
+        agent_name="Archie"
     )
-    
-    # Handle the agent connection with proper cleanup
-    try:
-        await _handle_agent_connection(websocket, realtime_agent, chat_history)
-    finally:
-        # Clean up MCP connection if it was established
-        if orders_api_plugin:
-            try:
-                await orders_api_plugin.disconnect()
-            except Exception:
-                pass
 
-
-async def _handle_agent_connection(websocket: WebSocket, realtime_agent, chat_history: ChatHistory):
-    """Handle the agent connection and audio streaming."""
     # Create the realtime client session
     async with realtime_agent(create_response=True) as client:
         # Start handling the messages from the realtime client with callback to forward the audio to acs
         receive_task = asyncio.create_task(
-            handle_realtime_messages(websocket, client, chat_history)
+            handle_realtime_messages(websocket, client, chat_history, is_development_mode)
         )
         # Receive messages from the ACS client and send them to the realtime client
         while True:

--- a/api/app/main.py
+++ b/api/app/main.py
@@ -55,39 +55,25 @@ async def agent_connect(websocket: WebSocket, acs_client=Depends(get_acs_client)
 
     chat_history = ChatHistory()
 
-    # Prepare plugins list
-    plugins = [
-        CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
-        DeliveryPlugin()
-    ]
-    
-    # Try to initialize MCP plugin, but fallback gracefully if it fails
-    orders_api_plugin = None
-    try:
-        # Attempt to create MCP plugin with a short timeout
-        orders_api_plugin = MCPStreamableHttpPlugin(
-            name="Orders API",
-            url=settings.MCP_ORDERS_URL,
-            load_prompts=False # APIM MCP servers only support tools and this will fail if left enabled
+    # Initialise MCP plugin with async context manager
+    async with MCPStreamableHttpPlugin(
+        name="Orders API",
+        url=settings.MCP_ORDERS_URL,
+        load_prompts=False # APIM MCP servers only support tools and this will fail if left enabled
+    ) as orders_api_plugin:
+
+        # Load realtime agent from template and plugins
+        realtime_agent = await get_agent(
+            template_name="DeliveryAgent",
+            plugins=[
+                CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
+                orders_api_plugin,
+                DeliveryPlugin()
+            ],
+            chat_history=chat_history,
+            agent_name="Ollie"
         )
-        # Try to connect with timeout
-        await asyncio.wait_for(orders_api_plugin.connect(), timeout=5.0)
-        plugins.insert(1, orders_api_plugin)  # Insert between CallPlugin and DeliveryPlugin
-        logger.info("Successfully connected to MCP Orders API")
-    except Exception as e:
-        logger.warning(f"Failed to connect to MCP Orders API: {e}. Continuing without orders functionality.")
-        orders_api_plugin = None
 
-    # Load realtime agent from template and plugins
-    realtime_agent = await get_agent(
-        template_name="DeliveryAgent",
-        plugins=plugins,
-        chat_history=chat_history,
-        agent_name="Ollie"
-    )
-
-    # Handle the agent connection with proper cleanup
-    try:
         # Create the realtime client session
         async with realtime_agent(create_response=True) as client:
             # Start handling the messages from the realtime client with callback to forward the audio to acs
@@ -118,13 +104,6 @@ async def agent_connect(websocket: WebSocket, acs_client=Depends(get_acs_client)
                     break
 
             receive_task.cancel()
-    finally:
-        # Clean up MCP connection if it was established
-        if orders_api_plugin:
-            try:
-                await orders_api_plugin.disconnect()
-            except Exception:
-                pass
 
 
 if __name__ == "__main__":

--- a/api/app/main.py
+++ b/api/app/main.py
@@ -52,55 +52,82 @@ async def agent_connect(websocket: WebSocket, acs_client=Depends(get_acs_client)
 
     chat_history = ChatHistory()
 
-    # Initialise MCP plugin with async context manager
-    async with MCPStreamableHttpPlugin(
-        name="Orders API",
-        url=settings.MCP_ORDERS_URL,
-        load_prompts=False # APIM MCP servers only support tools and this will fail if left enabled
-    ) as orders_api_plugin:
-
-        # Load realtime agent from template and plugins
-        realtime_agent = await get_agent(
-            template_name="DeliveryAgent",
-            plugins=[
-                CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
-                orders_api_plugin,
-                DeliveryPlugin()
-            ],
-            chat_history=chat_history,
-            agent_name="Ollie"
+    # Prepare plugins list
+    plugins = [
+        CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
+        DeliveryPlugin()
+    ]
+    
+    # Try to initialize MCP plugin, but fallback gracefully if it fails
+    orders_api_plugin = None
+    try:
+        # Attempt to create MCP plugin with a short timeout
+        orders_api_plugin = MCPStreamableHttpPlugin(
+            name="Orders API",
+            url=settings.MCP_ORDERS_URL,
+            load_prompts=False # APIM MCP servers only support tools and this will fail if left enabled
         )
+        # Try to connect with timeout
+        import asyncio
+        await asyncio.wait_for(orders_api_plugin.connect(), timeout=5.0)
+        plugins.insert(1, orders_api_plugin)  # Insert between CallPlugin and DeliveryPlugin
+        logger.info("Successfully connected to MCP Orders API")
+    except Exception as e:
+        logger.warning(f"Failed to connect to MCP Orders API: {e}. Continuing without orders functionality.")
+        orders_api_plugin = None
+    
+    # Load realtime agent from template and plugins
+    realtime_agent = await get_agent(
+        template_name="DeliveryAgent",
+        plugins=plugins,
+        chat_history=chat_history,
+        agent_name="Ollie"
+    )
+    
+    # Handle the agent connection with proper cleanup
+    try:
+        await _handle_agent_connection(websocket, realtime_agent, chat_history)
+    finally:
+        # Clean up MCP connection if it was established
+        if orders_api_plugin:
+            try:
+                await orders_api_plugin.disconnect()
+            except Exception:
+                pass
 
-        # Create the realtime client session
-        async with realtime_agent(create_response=True) as client:
-            # Start handling the messages from the realtime client with callback to forward the audio to acs
-            receive_task = asyncio.create_task(
-                handle_realtime_messages(websocket, client, chat_history)
-            )
-            # Receive messages from the ACS client and send them to the realtime client
-            while True:
-                try:
-                    # Receive data from the ACS client
-                    stream_data = await websocket.receive_text()
-                    data = json.loads(stream_data)
-                    logger.debug(f"Received data from ACS client: {data}")
 
-                    # If audio send it to the Realtime service
-                    if data["kind"] == "AudioData":
-                        await client.send(
-                            event=RealtimeAudioEvent(
-                                audio=AudioContent(
-                                    data=data["audioData"]["data"],
-                                    data_format="base64",
-                                    inner_content=data,
-                                ),
-                            )
+async def _handle_agent_connection(websocket: WebSocket, realtime_agent, chat_history: ChatHistory):
+    """Handle the agent connection and audio streaming."""
+    # Create the realtime client session
+    async with realtime_agent(create_response=True) as client:
+        # Start handling the messages from the realtime client with callback to forward the audio to acs
+        receive_task = asyncio.create_task(
+            handle_realtime_messages(websocket, client, chat_history)
+        )
+        # Receive messages from the ACS client and send them to the realtime client
+        while True:
+            try:
+                # Receive data from the ACS client
+                stream_data = await websocket.receive_text()
+                data = json.loads(stream_data)
+                logger.debug(f"Received data from ACS client: {data}")
+
+                # If audio send it to the Realtime service
+                if data["kind"] == "AudioData":
+                    await client.send(
+                        event=RealtimeAudioEvent(
+                            audio=AudioContent(
+                                data=data["audioData"]["data"],
+                                data_format="base64",
+                                inner_content=data,
+                            ),
                         )
-                except Exception:
-                    logger.info("Websocket connection closed.")
-                    break
+                    )
+            except Exception:
+                logger.info("Websocket connection closed.")
+                break
 
-            receive_task.cancel()
+        receive_task.cancel()
 
 
 if __name__ == "__main__":

--- a/api/app/main.py
+++ b/api/app/main.py
@@ -8,11 +8,13 @@ from pathlib import Path
 from fastapi import Depends, FastAPI, WebSocket
 from fastapi.responses import FileResponse
 from fastapi.staticfiles import StaticFiles
+from semantic_kernel.connectors.mcp import MCPStreamableHttpPlugin
 from semantic_kernel.contents import (AudioContent, ChatHistory,
                                       RealtimeAudioEvent)
 
 from .agents.plugins import CallPlugin, DeliveryPlugin
 from .agents.utils import get_agent, handle_realtime_messages
+from .config import settings
 from .dependencies import get_acs_client
 from .routers import calls
 
@@ -46,53 +48,83 @@ async def agent_connect(websocket: WebSocket, acs_client=Depends(get_acs_client)
     call_connection_id = websocket.headers.get("x-ms-call-connection-id")
 
     if not call_connection_id:
-        logger.warning("No call connection ID provided in headers indicating direct connection (not ACS). Certain call functions won't work.")
-    
+        logger.warning("No call connection ID provided in headers indicating direct connection (not ACS). Certain call functions will be mocked.")
+
     # Determine if running in development mode (for transcription safety)
     is_development_mode = not call_connection_id
-    
-    # Load realtime agent from template and plugins
+
     chat_history = ChatHistory()
+
+    # Prepare plugins list
+    plugins = [
+        CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
+        DeliveryPlugin()
+    ]
+    
+    # Try to initialize MCP plugin, but fallback gracefully if it fails
+    orders_api_plugin = None
+    try:
+        # Attempt to create MCP plugin with a short timeout
+        orders_api_plugin = MCPStreamableHttpPlugin(
+            name="Orders API",
+            url=settings.MCP_ORDERS_URL,
+            load_prompts=False # APIM MCP servers only support tools and this will fail if left enabled
+        )
+        # Try to connect with timeout
+        await asyncio.wait_for(orders_api_plugin.connect(), timeout=5.0)
+        plugins.insert(1, orders_api_plugin)  # Insert between CallPlugin and DeliveryPlugin
+        logger.info("Successfully connected to MCP Orders API")
+    except Exception as e:
+        logger.warning(f"Failed to connect to MCP Orders API: {e}. Continuing without orders functionality.")
+        orders_api_plugin = None
+
+    # Load realtime agent from template and plugins
     realtime_agent = await get_agent(
         template_name="DeliveryAgent",
-        plugins=[
-            CallPlugin(acs_client=acs_client, call_connection_id=call_connection_id),
-            DeliveryPlugin()
-        ],
+        plugins=plugins,
         chat_history=chat_history,
-        agent_name="Archie"
+        agent_name="Ollie"
     )
 
-    # Create the realtime client session
-    async with realtime_agent(create_response=True) as client:
-        # Start handling the messages from the realtime client with callback to forward the audio to acs
-        receive_task = asyncio.create_task(
-            handle_realtime_messages(websocket, client, chat_history, is_development_mode)
-        )
-        # Receive messages from the ACS client and send them to the realtime client
-        while True:
-            try:
-                # Receive data from the ACS client
-                stream_data = await websocket.receive_text()
-                data = json.loads(stream_data)
-                logger.debug(f"Received data from ACS client: {data}")
+    # Handle the agent connection with proper cleanup
+    try:
+        # Create the realtime client session
+        async with realtime_agent(create_response=True) as client:
+            # Start handling the messages from the realtime client with callback to forward the audio to acs
+            receive_task = asyncio.create_task(
+                handle_realtime_messages(websocket, client, chat_history, is_development_mode)
+            )
+            # Receive messages from the ACS client and send them to the realtime client
+            while True:
+                try:
+                    # Receive data from the ACS client
+                    stream_data = await websocket.receive_text()
+                    data = json.loads(stream_data)
+                    logger.debug(f"Received data from ACS client: {data}")
 
-                # If audio send it to the Realtime service
-                if data["kind"] == "AudioData":
-                    await client.send(
-                        event=RealtimeAudioEvent(
-                            audio=AudioContent(
-                                data=data["audioData"]["data"],
-                                data_format="base64",
-                                inner_content=data,
-                            ),
+                    # If audio send it to the Realtime service
+                    if data["kind"] == "AudioData":
+                        await client.send(
+                            event=RealtimeAudioEvent(
+                                audio=AudioContent(
+                                    data=data["audioData"]["data"],
+                                    data_format="base64",
+                                    inner_content=data,
+                                ),
+                            )
                         )
-                    )
-            except Exception:
-                logger.info("Websocket connection closed.")
-                break
+                except Exception:
+                    logger.info("Websocket connection closed.")
+                    break
 
-        receive_task.cancel()
+            receive_task.cancel()
+    finally:
+        # Clean up MCP connection if it was established
+        if orders_api_plugin:
+            try:
+                await orders_api_plugin.disconnect()
+            except Exception:
+                pass
 
 
 if __name__ == "__main__":

--- a/api/app/routers/calls.py
+++ b/api/app/routers/calls.py
@@ -17,11 +17,6 @@ from ..dependencies import get_acs_client
 router = APIRouter()
 logger = logging.getLogger(__name__)
 
-@router.post("/calls/test-validation")
-async def test_validation_handler():
-    """Test validation endpoint."""
-    return {"validationResponse": "test-success"}
-
 @router.post("/calls/incoming")
 async def incoming_call_handler(events: list[dict], acs_client=Depends(get_acs_client)):
     """Handle incoming call events from Azure Communication Services."""

--- a/api/app/routers/calls.py
+++ b/api/app/routers/calls.py
@@ -17,9 +17,16 @@ from ..dependencies import get_acs_client
 router = APIRouter()
 logger = logging.getLogger(__name__)
 
+@router.post("/calls/test-validation")
+async def test_validation_handler():
+    """Test validation endpoint."""
+    return {"validationResponse": "test-success"}
+
 @router.post("/calls/incoming")
 async def incoming_call_handler(events: list[dict], acs_client=Depends(get_acs_client)):
     """Handle incoming call events from Azure Communication Services."""
+    logger.info(f"Received {len(events)} events")
+    
     # This should be set in env when running locally to devtunnel uri
     if not settings.AZURE_ACS_CALLBACK_HOST_URI:
         # When running in container app, this will be the request host uri
@@ -39,16 +46,18 @@ async def incoming_call_handler(events: list[dict], acs_client=Depends(get_acs_c
     logger.info(f"Call event received. Using callback events URI: {callback_events_uri}")
   
     for event_dict in events:
-        event = EventGridEvent.from_dict(event_dict)
+        logger.info(f"Processing event: {event_dict.get('eventType', 'Unknown')}")
+        try:
+            event = EventGridEvent.from_dict(event_dict)
+            logger.info(f"Event parsed successfully: {event.event_type}")
 
-        match event.event_type:
-            case SystemEventNames.EventGridSubscriptionValidationEventName:
+            if event.event_type == SystemEventNames.EventGridSubscriptionValidationEventName:
                 logger.info("Validating subscription")
                 validation_code = event.data["validationCode"]
                 validation_response = {"validationResponse": validation_code}
+                logger.info(f"Returning validation response: {validation_response}")
                 return validation_response
-
-            case SystemEventNames.AcsIncomingCallEventName:
+            elif event.event_type == SystemEventNames.AcsIncomingCallEventName:
                 logger.debug("Incoming call received: data=%s", event.data)
                 caller_id = (
                     event.data["from"]["phoneNumber"]["value"]
@@ -84,9 +93,13 @@ async def incoming_call_handler(events: list[dict], acs_client=Depends(get_acs_c
                     ),
                 )
                 logger.info(f"Answered call for connection id: {answer_call_result.call_connection_id}")
-            case _:
+            else:
                 logger.debug("Event type not handled: %s", event.event_type)
                 logger.debug("Event data: %s", event.data)
+        except Exception as e:
+            logger.error(f"Error processing event: {e}")
+            import traceback
+            traceback.print_exc()
 
 @router.post("/calls/callbacks/{contextId}")
 async def callbacks(contextId: str, events: list[dict], acs_client=Depends(get_acs_client)):

--- a/api/app/static/index.html
+++ b/api/app/static/index.html
@@ -3,7 +3,7 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Voice Call Debugger</title>
+    <title>AI Contact Centre - Voice Call with Transcription</title>
     <style>
         body {
             font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
@@ -158,13 +158,56 @@
         .clear-btn:hover {
             background-color: #545b62;
         }
+        .transcription-container {
+            background-color: #f8f9fa;
+            border: 1px solid #dee2e6;
+            border-radius: 6px;
+            padding: 20px;
+            height: 300px;
+            overflow-y: auto;
+            font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
+            font-size: 14px;
+            line-height: 1.6;
+            margin-bottom: 20px;
+        }
+        .transcript-entry {
+            margin-bottom: 12px;
+            padding: 8px 12px;
+            border-radius: 8px;
+            max-width: 80%;
+        }
+        .transcript-entry.user {
+            background-color: #e3f2fd;
+            border-left: 4px solid #2196f3;
+            margin-left: auto;
+            text-align: right;
+        }
+        .transcript-entry.assistant {
+            background-color: #f3e5f5;
+            border-left: 4px solid #9c27b0;
+            margin-right: auto;
+        }
+        .transcript-speaker {
+            font-weight: bold;
+            font-size: 12px;
+            color: #666;
+            margin-bottom: 4px;
+        }
+        .transcript-text {
+            color: #333;
+        }
+        .transcript-timestamp {
+            font-size: 10px;
+            color: #999;
+            margin-top: 4px;
+        }
     </style>
 </head>
 <body>
     <div class="container">
         <div class="header">
-            <h1>Voice Call Debugger</h1>
-            <p>Test WebSocket audio streaming with microphone input</p>
+            <h1>AI Contact Centre - Voice Call with Transcription</h1>
+            <p>Test WebSocket audio streaming with microphone input and live transcription</p>
         </div>
 
         <div id="status" class="status disconnected">
@@ -175,9 +218,15 @@
             <button id="startBtn" class="btn btn-primary">Start Call</button>
             <button id="stopBtn" class="btn btn-danger" disabled>Stop Call</button>
             <button id="clearLogBtn" class="btn btn-secondary">Clear Log</button>
+            <button id="clearTranscriptBtn" class="btn btn-secondary">Clear Transcript</button>
         </div>
 
         <!-- Audio visuals removed as requested -->
+
+        <div>
+            <h3>Live Transcription</h3>
+            <div id="transcriptContainer" class="transcription-container"></div>
+        </div>
 
         <div>
             <h3>Event Log</h3>
@@ -207,8 +256,10 @@
                 this.startBtn = document.getElementById('startBtn');
                 this.stopBtn = document.getElementById('stopBtn');
                 this.clearLogBtn = document.getElementById('clearLogBtn');
+                this.clearTranscriptBtn = document.getElementById('clearTranscriptBtn');
                 this.statusDiv = document.getElementById('status');
                 this.logContainer = document.getElementById('logContainer');
+                this.transcriptContainer = document.getElementById('transcriptContainer');
                 
                 // Audio visuals removed
             }
@@ -217,6 +268,7 @@
                 this.startBtn.addEventListener('click', () => this.startCall());
                 this.stopBtn.addEventListener('click', () => this.stopCall());
                 this.clearLogBtn.addEventListener('click', () => this.clearLog());
+                this.clearTranscriptBtn.addEventListener('click', () => this.clearTranscript());
             }
 
             log(message, type = 'info') {
@@ -231,6 +283,37 @@
             clearLog() {
                 this.logContainer.innerHTML = '';
                 this.log('Log cleared', 'info');
+            }
+
+            clearTranscript() {
+                this.transcriptContainer.innerHTML = '';
+                this.log('Transcript cleared', 'info');
+            }
+
+            addTranscript(speaker, text, timestamp = null) {
+                const transcriptEntry = document.createElement('div');
+                transcriptEntry.className = `transcript-entry ${speaker}`;
+                
+                const speakerLabel = document.createElement('div');
+                speakerLabel.className = 'transcript-speaker';
+                speakerLabel.textContent = speaker === 'user' ? 'You' : 'AI Assistant';
+                
+                const transcriptText = document.createElement('div');
+                transcriptText.className = 'transcript-text';
+                transcriptText.textContent = text;
+                
+                transcriptEntry.appendChild(speakerLabel);
+                transcriptEntry.appendChild(transcriptText);
+                
+                if (timestamp !== null) {
+                    const timestampDiv = document.createElement('div');
+                    timestampDiv.className = 'transcript-timestamp';
+                    timestampDiv.textContent = `${timestamp}ms`;
+                    transcriptEntry.appendChild(timestampDiv);
+                }
+                
+                this.transcriptContainer.appendChild(transcriptEntry);
+                this.transcriptContainer.scrollTop = this.transcriptContainer.scrollHeight;
             }
 
             updateStatus(status, className) {
@@ -391,6 +474,12 @@
                 
                 if (data.kind === "AudioData") {
                     this.playAudioData(data.audioData.data);
+                } else if (data.kind === "Transcription") {
+                    const { speaker, text, timestamp } = data.data;
+                    this.addTranscript(speaker, text, timestamp);
+                    this.log(`Transcription [${speaker}]: ${text}`, 'info');
+                } else if (data.kind === "ChatHistory") {
+                    this.log(`Chat history updated: ${data.data.length} messages`, 'info');
                 }
             }
 

--- a/api/app/static/index.html
+++ b/api/app/static/index.html
@@ -3,7 +3,7 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>AI Contact Centre - Voice Call with Transcription</title>
+    <title>Voice Call Debugger</title>
     <style>
         body {
             font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
@@ -206,8 +206,8 @@
 <body>
     <div class="container">
         <div class="header">
-            <h1>AI Contact Centre - Voice Call with Transcription</h1>
-            <p>Test WebSocket audio streaming with microphone input and live transcription</p>
+            <h1>Voice Call Debugger</h1>
+            <p>Test WebSocket audio streaming with microphone input</p>
         </div>
 
         <div id="status" class="status disconnected">

